flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -406,7 +422,8 @@ void createIcebergTable(ObjectPath tablePath, ResolvedCatalogTable table, boolea
     for (Map.Entry<String, String> entry : table.getOptions().entrySet()) {
       if ("location".equalsIgnoreCase(entry.getKey())) {
         location = entry.getValue();
-      } else {
+      } else if (!("connector".equalsIgnoreCase(entry.getKey())
+          || FlinkCreateTableOptions.SRC_CATALOG_PROPS_KEY.equalsIgnoreCase(entry.getKey()))) {
```

Comment:
It took me some time to decipher this 😄 
Could we please break up the negation so it doesn't span multiple lines?
Maybe:
```
      } else if (!"connector".equalsIgnoreCase(entry.getKey())
          && !FlinkCreateTableOptions.SRC_CATALOG_PROPS_KEY.equalsIgnoreCase(entry.getKey())) {
```

Or it is just me?
Also maybe a comment should be nice